### PR TITLE
handle duplicate pubkeys in sync committee

### PR DESCRIPTION
**Pull request by etan-status**
When sync committee message handling was introduced in #2830, the edge
case of the same validator being selected multiple times as part of a
sync subcommittee was not covered. Not handling that edge case makes
sync contributions have a lower-than-expected participation rate as each
sync validator is only counted up through once per subcommittee.
This patch ensures that this edge case is properly covered.

### CODE REVIEW

**Comment by etan-status** [comment]
The new test fails to run on Windows, investigating.

```
[Suite] Extra
all_tests --xml:build/all_tests.xml --console failed; Aborting.
mingw32-make: *** [Makefile:257: test] Error 1
Error: Process completed with exit code 2.
```

**Comment by etan-status** [comment]
Windows problem was due to a missing `newClone`. Fixed now.

File: tests/test_gossip_validation.nim [hunk 1]
```diff
@@ -170,3 +174,68 @@ suite "Gossip validation " & preset():
...
+            int(SLOTS_PER_EPOCH), false, cfg = cfg):
+          let added = 
+            case blck.kind
+            of BeaconBlockFork.Phase0:
+              const nilCallback = OnPhase0BlockAdded(nil)
+              dag.addRawBlock(quarantine, blck.phase0Block, nilCallback)
+            of BeaconBlockFork.Altair:
+              const nilCallback = OnAltairBlockAdded(nil)
+              dag.addRawBlock(quarantine, blck.altairBlock, nilCallback)
+          doAssert added.isOk()
```
Comment (tersec) [inline tests/test_gossip_validation.nim]: `check: added.isOk()` is better when running in a testing context

**Review by zah — APPROVED** [review]

**Review by tersec — APPROVED** [review]

File: tests/test_gossip_validation.nim (hunk 1 above)
Comment (etan-status) [reply tests/test_gossip_validation.nim]: OK, also changing the other instance of `doAssert` further down in the test.
